### PR TITLE
Simplify some unit tests

### PR DESCRIPTION
**Pull request by loiclec**
# Pull Request

## What does this PR do?
Simplify the code that is used in unit tests to create and modify an index. Basically, the following code:
```rust
  let path = tempfile::tempdir().unwrap();
  let mut options = EnvOpenOptions::new();
  options.map_size(10 * 1024 * 1024); // 10 MB
  let index = Index::new(options, &path).unwrap();

  let mut wtxn = index.write_txn().unwrap();
  let content = documents!([
      { "id": 0, "name": "kevin" },
  ]);
  let config = IndexerConfig::default();
  let indexing_config = IndexDocumentsConfig::default();
  let builder =
      IndexDocuments::new(&mut wtxn, &index, &config, indexing_config, |_| ()).unwrap();
  let (builder, user_error) = builder.add_documents(content).unwrap();
  user_error.unwrap();
  builder.execute().unwrap();
  wtxn.commit.unwrap();

  let mut wtxn = index.write_txn().unwrap();
  let config = IndexerConfig::default();
  let mut builder = Settings::new(&mut wtxn, &index, &config);
  builder.set_primary_key(S("docid"));
  builder.set_filterable_fields(hashset! { S("label") });
  builder.execute(|_| ()).unwrap();
  wtxn.commit().unwrap();
```
becomes:
```rust
let index = TempIndex::new():
index.add_documents(documents!(
      { "id": 0, "name": "kevin" },
)).unwrap();
index.update_settings(|settings| {
    settings.set_primary_key(S("docid"));
    settings.set_filterable_fields(hashset! { S("label") });
}).unwrap();
```

Then there is a bunch of options to modify the indexing configs, the map size, to reuse a transaction, etc. For example:
```rust
let mut index = TempIndex::new_with_map_size(1000 * 4096 * 10);
index.index_documents_config.autogenerate_docids = true;
let mut wtxn = index.write_txn().unwrap();
index.update_settings_using_wtxn(&mut wtxn, |settings| {
    settings.set_primary_key(S("docids"));
}).unwrap();
wtxn.commit().unwrap();
```

### CODE REVIEW

File: milli/src/index.rs [hunk 1]
```diff
@@ -1185,43 +1188,88 @@ pub(crate) mod tests {
...
+        pub fn update_settings(
+            &self,
+            update: impl Fn(&mut Settings),
+        ) -> Result<(), crate::error::Error> {
+            let mut wtxn = self.write_txn().unwrap();
+            self.update_settings_using_wtxn(&mut wtxn, update)?;
+            wtxn.commit().unwrap();
+            Ok(())
+        }
+        pub fn update_settings_using_wtxn<'t>(
```
Comment (ManyTheFish) [inline milli/src/index.rs:1249]:
```suggestion
        }
        
        pub fn update_settings_using_wtxn<'t>(
```
did you configure rustfmt on your editor?

File: milli/src/index.rs [hunk 2]
```diff
@@ -1185,43 +1188,88 @@ pub(crate) mod tests {
...
+            let indexer_config = IndexerConfig::default();
+            let index_documents_config = IndexDocumentsConfig::default();
+            Self { inner, indexer_config, index_documents_config, _tempdir }
+        }
+        /// Creates a temporary index, with a default `4096 * 1000` size. This should be enough for
+        /// most tests.
+        pub fn new() -> Self {
+            Self::new_with_map_size(4096 * 1000)
+        }
+        pub fn add_documents_using_wtxn<'t, R>(
```
Comment (ManyTheFish) [inline milli/src/index.rs:1206]:
```suggestion
        }

        pub fn add_documents_using_wtxn<'t, R>(
```

**Review by ManyTheFish — CHANGES_REQUESTED** [review]
Nice PR -619 lines!

I just requested some formatting changes.

Could you rebase your branch, please? Unfortunately, I added a new test recently that does not follow your new test structure 😬

File: milli/src/index.rs (hunk 1 above)
Comment (loiclec) [reply milli/src/index.rs:1249]: I did, but for some reason every once in a while it doesn't work for a file :/

**Review by irevoire — APPROVED** [review]
noice

**Review by ManyTheFish — APPROVED** [review]
Thank you @loiclec,

bors merge

**Pull request merged**